### PR TITLE
MM Scenario improvements, next part

### PR DESCRIPTION
**Pull request by SJuliez**
This adds ammo count and Mek crit parsing to the scenario yaml definition language and adds a another Kell Hounds scenario (Death of the Legion).

### CODE REVIEW

File: megamek/data/scenarios/Kell Hounds/DeathOfTheLegion/DeathOfTheLegion.mms [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+  camo: Mercs/Kell Hounds.jpg
+  deploy: W
+
+  units:
+    - fullname: Thunderbolt TDR-5S
+      id: 101
+      force: Command Lance|1
+      crew:
+        name: Lt. Col. Patrick M. Kell
+        portrait: Male/MechWarrior/MW_M_27.png
```
Comment (rjhancock) [inline megamek/data/scenarios/Kell Hounds/DeathOfTheLegion/DeathOfTheLegion.mms:49]: Correct usage needed (and elsewhere).

File: megamek/data/scenarios/Kell Hounds/LoweringTheBoom/LoweringTheBoom.mms [hunk 2]
```diff
@@ -59,6 +59,12 @@ factions:
...
         CT: 15
+    ammo:
+      CT:
+        - slot: 11
+          shots: 5
+        - slot: 12
+          shots: 5
     crew:
       name: Col. Oliver Nage
       portrait: Male/MechWarrior/MW_M_15.png
```
Comment (rjhancock) [inline megamek/data/scenarios/Kell Hounds/LoweringTheBoom/LoweringTheBoom.mms:70]: Correct Naming (and elsewhere in file).

File: megamek/docs/Scenarios/ScenarioV2 HowTo.mms [hunk 3]
```diff
@@ -208,11 +219,52 @@ factions:
...
+        # all fields in crew are optional
+        crew:
           name: Cpt. Frederic Nguyen
+          callsign: MAGIC
           piloting: 4
           gunnery: 3
+          # Optional: pilot hits, 0 to 6
+          hits: 3
+          # Optional: a portrait, relative to data/images/portraits
+          portrait: Male/MechWarrior/MW_M_13.png
```
Comment (rjhancock) [inline megamek/docs/Scenarios/ScenarioV2 HowTo.mms:267]: Here to.

File: megamek/src/megamek/common/jacksonadapters/EntityDeserializer.java [hunk 4]
```diff
@@ -244,4 +265,79 @@ private void assignVelocity(Entity entity, JsonNode node) {
             ((IAero) entity).setNextVelocity(velocity);
         }
     }
-}
\ No newline at end of file
+
+    private void assignCrits(Entity entity, JsonNode node) {
+        if (!(entity instanceof Mech) || !node.has(CRITS)) {
```
Comment (rjhancock) [inline megamek/src/megamek/common/jacksonadapters/EntityDeserializer.java:270]: Correct class name and add JavaDoc comments to all added methods.

File: megamek/data/scenarios/Kell Hounds/DeathOfTheLegion/DeathOfTheLegion.mms (hunk 1 above)
Comment (SJuliez) [reply megamek/data/scenarios/Kell Hounds/DeathOfTheLegion/DeathOfTheLegion.mms:49]: done

File: megamek/data/scenarios/Kell Hounds/LoweringTheBoom/LoweringTheBoom.mms (hunk 2 above)
Comment (SJuliez) [reply megamek/data/scenarios/Kell Hounds/LoweringTheBoom/LoweringTheBoom.mms]: done

**Review by neoancient — APPROVED** [review]

**Pull request merged**